platforms/android/demo/src/main/java/com/mapzen/tangram/android/MainActivity.java

```diff
@@ -31,36 +40,130 @@
+                        notifyDataSetChanged();
+                    }
+                }
+            }
+        };
+
+        CustomArrayAdapter(Context context, int resource, List<String> objects) {
+            super(context, resource, objects);
+            suggestions = new ArrayList<>();
+        }
+
+        @Override public Filter getFilter() {
+            return nameFilter;
+        }
+    }
 
     MapController map;
     MapView view;
     LngLat lastTappedPoint;
     MapData markers;
 
+    private final String[] scenes = {
+        "scene.yaml",
+        "https://mapzen.com/carto/bubble-wrap-style/bubble-wrap-style.zip",
+        "https://mapzen.com/carto/refill-style/refill-style.zip",
+        "https://mapzen.com/carto/walkabout-style/walkabout-style.zip",
+        "https://mapzen.com/carto/tron-style/tron-style.zip",
+        "https://mapzen.com/carto/cinnabar-style/cinnabar-style.zip",
+        "https://mapzen.com/carto/zinc-style/zinc-style.zip"
+    };
+
+    AutoCompleteTextView autoCompleteView;
+    CustomArrayAdapter customAdapter;
+    Spinner spinner;
+
     String pointStylingPath = "layers.touch.point.draw.icons";
     ArrayList<Marker> pointMarkers = new ArrayList<Marker>();
 
     boolean showTileInfo = false;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
-        final ArrayList<SceneUpdate> sceneUpdates = new ArrayList<>(1);
-        final String apiKey = "vector-tiles-tyHL4AY";
         sceneUpdates.add(new SceneUpdate("global.sdk_mapzen_api_key", apiKey));
         super.onCreate(savedInstanceState);
 
         requestWindowFeature(Window.FEATURE_NO_TITLE);
 
         setContentView(R.layout.main);
 
+        /* setup autocompletetextview adapter */
+        autoCompleteView = (AutoCompleteTextView)findViewById(R.id.autoCompleteTextView);
+        List<String> lst = new ArrayList<String>(Arrays.asList(scenes));
+        customAdapter = new CustomArrayAdapter(this, android.R.layout.simple_list_item_1, lst);
+        autoCompleteView.setAdapter(customAdapter);
+        autoCompleteView.setThreshold(1);
+        autoCompleteView.setOnItemClickListener(this);
+
+        /* setup spinner style selecter */
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayShowTitleEnabled(false);
+        spinner = (Spinner) findViewById(R.id.spinner);
+        ArrayAdapter<CharSequence> sceneSelector =
+                ArrayAdapter.createFromResource(this, R.array.style_array, R.layout.simple_spinner_item);
+        sceneSelector.setDropDownViewResource(R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(sceneSelector);
+        spinner.setOnItemSelectedListener(this);
+
+        /* setup map view */
         view = (MapView)findViewById(R.id.map);
         view.onCreate(savedInstanceState);
+        autoCompleteView.setText(scenes[0]);
         view.getMapAsync(this, "scene.yaml", sceneUpdates);
```

Comment:
This call should use the current text in the `AutoCompleteView` instead of a constant.